### ISSUE TITLE
apt-key is deprecated

### ISSUE THREAD

**Issue by hholst80** [NVIDIA/k8s-device-plugin#369]
`curl ... | apt-key add -` is deprecated. The suggested workflow is to add the gpg key directly to the file system.

tl;dr store it in `/etc/apt/keyrings/libnvidia-container.gpg` (if placed there by OS admin) and point it out in .list file with `[signed-by=/etc/apt/keyrings/libnvidia-container.gpg]` as sketched [here](https://nvidia.github.io/libnvidia-container/)

apt-key manual states:

<blockquote>
Then you can directly replace this with (though note the recommendation below):

       wget -qO- https://myrepo.example/myrepo.asc | sudo tee /etc/apt/trusted.gpg.d/myrepo.asc

Make sure to use the "asc" extension for ASCII armored keys and the "gpg" extension for the binary OpenPGP
format (also known as "GPG key public ring"). The binary OpenPGP format works for all apt versions, while the
ASCII armored format works for apt version >= 1.4.

Recommended: Instead of placing keys into the /etc/apt/trusted.gpg.d directory, you can place them anywhere
on your filesystem by using the Signed-By option in your sources.list and pointing to the filename of the
key. See sources.list(5) for details. Since APT 2.4, /etc/apt/keyrings is provided as the recommended
location for keys not managed by packages. When using a deb822-style sources.list, and with apt version >=
2.4, the Signed-By option can also be used to include the full ASCII armored keyring directly in the
sources.list without an additional file.
</blockquote>

**Comment by stanislavvv** [comment]
Agree with issue — apt-key deprecated and must not used.
I don't see how to fix https://nvidia.github.io/libnvidia-container/$distribution/libnvidia-container.list — it's not in this repo and must be fixed separatedly (add `[signed-by=/etc/apt/trusted.gpg.d/libnvidia-container.asc] ` after `deb ` in every repo line).
But in README section "Install the `nvidia-container-toolkit`" must be:

```bash
distribution=$(. /etc/os-release;echo $ID$VERSION_ID)
curl -s -L https://nvidia.github.io/libnvidia-container/gpgkey | sudo tee /etc/trusted.gpg.d/libnvidia-container.asc
curl -s -L https://nvidia.github.io/libnvidia-container/$distribution/libnvidia-container.list | sudo tee /etc/apt/sources.list.d/libnvidia-container.list

sudo apt-get update && sudo apt-get install -y nvidia-container-toolkit
```

**Comment by elezar** [comment]
@stanislavvv the documentation here: https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html#installing-with-apt no longer uses `apt-key`.

I see that we have not updated the README and I will to so.
